### ISSUE TITLE
Newbie pnpm

### ISSUE THREAD

**Issue by xania** [pnpm/pnpm#6532]
`pnpm install` fails with error when name of package contains a '.'. Needless to say there lays the workaround but it took me days to figure this out. Can we inprove docs (add warning for allowed chars) or even better support the '.' chars in the package name and folder names.

### pnpm version:
8.5.0

### Code to reproduce the issue:
https://github.com/xania/xania.nl/tree/pnpm-repro-bug

### Expected behavior:
package deps are installed

### Actual behavior:

```shell
 WARN  The "workspaces" field in package.json is not supported by pnpm. Create a "pnpm-workspace.yaml" file instead.
 ERR_PNPM_NO_MATCHING_VERSION_INSIDE_WORKSPACE  In : No matching version found for xania.functions@* inside the workspace

This error happened while installing a direct dependency of C:\dev\xania.nl
Progress: resolved 1, reused 0, downloaded 0, added 0
```

### Additional information:

node -v
v19.3.0

OS: windows

**Comment by zkochan** [comment]
You need to create a [pnpm workspace](https://pnpm.io/workspaces).

Create a `pnpm-workspace.yaml` file in the root of the repository with this content:

```yaml
packages:
  - 'packages/*'

```

**Comment by xania** [comment]
> You need to create a [pnpm workspace](https://pnpm.io/workspaces).
> 
> Create a `pnpm-workspace.yaml` file in the root of the repository with this content:
> 
> ```yaml
> packages:
>   - 'packages/*'
> ```

I feel incredibly stupid. In my actual project I tested with fhe following configuration while seeing the error. I stripped out the most of the code while keeping the error and ended up with the code shown above. But if I try to reproduce it now then no error. Is there a possible exaplanation except me being stupid 🤒 ? cache maybe?

packages:
  - 'packages/*'
  - '!**/test/**'
